src/sdk_codes.ts

```diff
@@ -0,0 +1,171 @@
+import { Logger } from './log';
+import { Target } from './types';
+
+interface SDKCodeMessages {
+  [key: number]: string;
+}
+
+const sdkCodes: SDKCodeMessages = {
+  // SDK_INIT_1xxx
+  1000: 'The SDK has successfully initialized',
+  1001: 'The SDK has failed to initialize due to an authentication error - defaults will be served',
+  1002: 'The SDK has failed to initialize due to a missing or empty API key - defaults will be served',
+  // SDK_AUTH_2xxx
+  2000: 'Authentication was successful',
+  2001: 'Authentication failed with a non-recoverable error',
+  2002: 'Authentication attempt',
+  2003: 'Authentication failed and max retries have been exceeded',
+  // SDK_CLOSE_3xxx
+  3000: 'Closing SDK',
+  3001: 'SDK Closed successfully',
+  // SDK_POLL_4xxx
+  4000: 'Polling started, intervalMs:',
+  4001: 'Polling stopped, reason:',
+  // SDK_STREAM_5xxx
+  5000: 'SSE stream successfully connected',
+  5001: 'SSE stream disconnected, reason:',
+  5002: 'SSE event received: ',
+  5003: 'SSE retrying to connect in',
+  5004: 'SSE stopped',
+  // SDK_EVAL_6xxx -
+  6000: 'Evaluation successful: ',
+  6001: 'Evaluation Failed, returning default variation: ',
+  // SDK_METRICS_7xxx
+  7000: 'Metrics thread started with request interval:',
+  7001: 'Metrics stopped',
+  7002: 'Posting metrics failed, reason:',
+  7003: 'Metrics posted successfully',
+};
+
+export function getSDKCodeMessage(key: number): string {
+  if (key in sdkCodes) {
+    return sdkCodes[key];
+  } else {
+    return 'Unknown SDK code';
+  }
+}
+
+export function sdkErrMsg(errorCode: number, appendText = ''): string {
+  return `SDKCODE:${errorCode}: ${getSDKCodeMessage(errorCode)} ${appendText}`;
+}
+
+export function warnMissingSDKKey(logger: Logger): void {
+  logger.warn(sdkErrMsg(1002, ''));
```

Comment:
`appendText` is optional, so you can omit the blank string
```suggestion
  logger.warn(sdkErrMsg(1002));
```